src/Fmt_ast.ml

```diff
@@ -1077,9 +1077,21 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext
         $ fmt_if_k (not last) (break_unless_newline 1 0)
       in
       let is_nested_diff_prec_infix_ops =
+        let skip_parent_for_some_infix x =
+          match x.pexp_desc with
+          | Pexp_ident {txt= Lident i} -> (
+            match i with
+            | "==" | "=" | "<" | ">" | "<=" | ">=" | "<>" | "!=" -> true
+            | ":=" -> true
+            | _ -> false )
+          | _ -> assert false
```

Comment:
If it doesn't complicate things, I'd prefer to avoid [assert false] and move the [is_infix] check here. I expect that would be clearer.